index.bs

```diff
@@ -488,21 +493,24 @@ Patch</h5>
 When the response is received at the end of a navigation, fill the [=bounce tracking record/bounce set=].
 
 Insert the following steps in the <a spec="html">navigate</a> algorithm, under "Create navigation params by fetching", after Step 19.7,
-"Wait until either response is non-null...", or as soon as [=response/url list=] becomes available.
+"Wait until either response is non-null...", as soon as [=response/url list=] becomes available.
 
 1. Run [=process response received for bounce tracking=] given
     <var ignore>navigable</var> and <var ignore>response</var>'s <var ignore>URL list</var>.
 
 <div algorithm>
 
 To <dfn>process response received for bounce tracking</dfn> given a [=navigable=]
-|navigable| and a [=response/url list=] |URLs|, perform the following steps:
+|navigable| and a [=list=] of [=URLs=] |URLs|, perform the following steps:
 
-1. [=Assert=] that |navigable|'s [=top-level traversable/bounce tracking record=]'s
-    [=bounce tracking record/client bounce detection running=] is false.
 1. If |navigable| is not a [=top-level traversable=], then abort these steps.
-1. Run [=run steps after a timeout=] given <var ignore>milliseconds</var> [=client bounce detection timer period=]
-    and <var ignore>completionSteps</var> [=record stateful bounces for bounce tracking=] given |navigable|.
+1. [=Assert=]: |navigable|'s [=top-level traversable/bounce tracking record=] is not null.
+1. Run [=run steps after a timeout=] given:
+    1. <var ignore>global</var> = |navigable|'s [=navigable/active document=]'s [=relevant global object=]
+    1. <var ignore>milliseconds</var> = [=client bounce detection timer period=]
+    1. <var ignore>completionSteps</var> = [=record stateful bounces for bounce tracking=] given |navigable|.
```

Comment:
The [completion steps](https://html.spec.whatwg.org/multipage/timers-and-user-prompts.html#run-steps-after-a-timeout) run in parallel, so this should queue a task to [=record stateful bounces...=]. Otherwise the  'bounce tracking record' could become null in the middle of another algorithm.

Reply:
Thanks. It is enough to state "queue a [=task=] to [=record stateful bounces...=]", or should we define a task queue to the Global Data?